tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/ServiceRegistrations.cs

```diff
@@ -115,19 +116,46 @@ public static void RegisterCommonServices(IServiceCollection services, OutputHel
                 // For more information about this pattern: https://learn.microsoft.com/en-us/dotnet/azure/sdk/dependency-injection
                 var service = new AzureService();
                 clientBuilder.UseCredential(service.GetCredential());
+            });
 
-                // Azure OpenAI client does not, for some reason, have an
-                // in-package facade for this, so register manually.
-                clientBuilder.AddClient<AzureOpenAIClient, AzureOpenAIClientOptions>(
-                    (options, credential, _) =>
-                    {
-                        var endpointEnvVar = Environment.GetEnvironmentVariable("AZURE_OPENAI_ENDPOINT");
-                        var ep = string.IsNullOrWhiteSpace(endpointEnvVar) ?
-                            "https://openai-shared.openai.azure.com"
-                            : endpointEnvVar;
+            // Register OpenAI client with endpoint and authentication
+            services.AddSingleton<OpenAIClient>(sp =>
+            {
+                var azureService = sp.GetRequiredService<IAzureService>();
+                var credential = azureService.GetCredential();
 
-                        return new AzureOpenAIClient(new Uri(ep), credential, options);
-                    });
+                var openAiApiKey = Environment.GetEnvironmentVariable("OPENAI_API_KEY");
```

Comment:
How do you expect these env variables to get setup?

Reply:
We discussed offline. These env variables are only to override the default credentials.